added uc8151d display controller support

mespunaiot opened the pull request:
## Summary

  - Extended the UC81xx SPI EPD driver with UC8151D-specific handling (packed TRES/PTL payloads, CDI overrides, and the extra PON-after-BTST quirk defined in the datasheet)
  - Added the UC8151D quirk table and hook it into the DT instantiation path alongside the existing UC8175/76/79 variants
  - Documented the new controller via ultrachip,uc8151d.yaml and update Kconfig dependencies so the driver enables when the new compatible appears

  ## Testing

  - Tested using the GDEW029I6FD from Good Display and a custom nrf9151 board.

mespunaiot committed:
drivers: display: uc8151d: Add UC8151D display controller support

Add support for the UltraChip UC8151D e-paper display (EPD) controller.
The UC8151D is part of the UC81xx family of display controllers commonly
used in e-ink displays.

This implementation extends the existing UC81xx driver infrastructure by
adding device tree bindings, Kconfig options, and the necessary driver
code to support the UC8151D variant.

Signed-off-by: Marc Espuña <mespuna@cactusiot.com>

carlescufi commented:
@VynDragon would you mind taking a look please?

VynDragon reviewed (approved):
I dont see anything wrong in this PR, a lot of debug logging and comments that feels sometimes unnecessary, but nothing i consider a issue.

josuah commented on drivers/display/uc81xx.c:
```diff
@@ -227,6 +229,22 @@ static int uc81xx_set_profile(const struct device *dev,
+			 * power initialization
+			 */
+			LOG_DBG("Sending PON command after softstart");
+			if (uc81xx_write_cmd(dev, UC81XX_CMD_PON, NULL, 0)) {
+				return -EIO;
+			}
```
`if (xxx) return -EIO;` does not return the correct error code from the driver like `err = xxx; if (err < 0) return err;` would do, but it's already like that everywhere else, so tuning it would be a different PR.

josuah reviewed (dismissed):
That's the best improvements I could propose but these are not problems to fix.

From @VynDragon I learned it's grayscale or red/black/white, though it's same number of bit-per-pixel declared in the driver (MONO01). I suppose it's just not enabled?

VynDragon reviewed (changes requested):
Forgot something, you need to add one entry to tests/drivers/build_all/display/app.yaml, use a separate commit.

VynDragon commented:
> I learned it's grayscale or red/black/white, though it's same number of bit-per-pixel declared in the driver (MONO01). I suppose it's just not enabled?

Or b&w.

mespunaiot committed:
tests: drivers: build_all: cover uc8151d

Add a UC8151D node to the build_all display overlay so the
continuous-integration build exercises the new driver path. The dummy
profile overrides only CDI and VCOM to keep the overlay lightweight
while still covering the relevant code paths.

Signed-off-by: Marc Espuña <mespuna@cactusiot.com>

mespunaiot commented on tests/drivers/build_all/display/app.overlay:
```diff
@@ -273,7 +273,28 @@
+				};
+			};
 		};
+	};
```
Right! Sorry, i'll fix that now
